### DISCUSSION TITLE
Destroying nested forms

### DISCUSSION THREAD

**Discussion by andrewr224** [rootstrap/yaaf#63] [Q&A] [answered]
How do you destroy nested models with yaaf?

**Comment by santib** [comment, marked as answer]
Hi @andrewr224, sorry for the late response. Currently, we don't have that feature. Thinking on the possibilities to implement it, I'm not sure what would be best:
1. Make the `save` method smarter by using [mark_for_destruction](https://api.rubyonrails.org/classes/ActiveRecord/AutosaveAssociation.html#method-i-mark_for_destruction) and the marked models gets deleted upon save.. The good thing is that it will allow creating + updating + deleting as part of the same form
2. Have `YAAF::Form` have a `destroy` method which destroys everything in the `@models` array
3. Have a new `YAAF::DestroyForm` which `save` method destroys everything in the `@models` array

Thoughts? cc @juanmanuelramallo

**Reply by juanmanuelramallo** [reply]
I'd use a before save callback and manually destroy whatever it's needed to be destroyed. 

I'm not sure if yaaf has much of a say for "destroying nested models" because yaaf is only an object that maps to a form, its fields, validations and side effects (callbacks). So perhaps destroying nested models might fit into the callbacks part of yaaf (?

If you don't need any special customization or data validation in the form, then I'd suggest using the AcceptsNestedAtrributesAPI that handles the destroy out of the box.
